Documentation/security/policy/kubernetes.rst

```diff
@@ -16,25 +16,41 @@ Namespaces
 
 `Namespaces <https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/>`_
 are used to create virtual clusters within a Kubernetes cluster. All Kubernetes objects
-including NetworkPolicy and CiliumNetworkPolicy belong to a particular
-namespace. Depending on how a policy is being defined and created, Kubernetes
-namespaces are automatically being taken into account:
-
-* Network policies created and imported as `CiliumNetworkPolicy` CRD and
-  `NetworkPolicy` apply within the namespace, i.e. the policy only applies
-  to pods within that namespace. It is however possible to grant access to and
-  from pods in other namespaces as described below.
-
-* Network policies imported directly via the :ref:`api_ref` apply to all
-  namespaces unless a namespace selector is specified as described below.
-
-.. note:: While specification of the namespace via the label
-	  ``k8s:io.kubernetes.pod.namespace`` in the ``fromEndpoints`` and
-	  ``toEndpoints`` fields is deliberately supported. Specification of the
-	  namespace in the ``endpointSelector`` is prohibited as it would
-	  violate the namespace isolation principle of Kubernetes. The
-	  ``endpointSelector`` always applies to pods of the namespace which is
-	  associated with the CiliumNetworkPolicy resource itself.
+including `NetworkPolicy` and `CiliumNetworkPolicy` belong to a particular
+namespace.
+
+Known Pitfalls
+~~~~~~~~~~~~~~
+
+When you work with namespaces within policies, be aware of some known pitfalls:
+
+* Depending on how a policy is defined and created, Kubernetes
+  namespaces are automatically taken into account:
+
+  * Network policies imported directly with the :ref:`api_ref` apply to all
+    namespaces unless a namespace selector is specified as described in
+    :ref:`example_cnp_ns_boundaries`.
+
+  * Network policies created and imported as `CiliumNetworkPolicy` CRD and
+    `NetworkPolicy` apply within the namespace; in other words, the policy **only** applies
+    to pods within that namespace. It's possible, however, to grant access to and
+    from pods in other namespaces as described in :ref:`example_cnp_across_ns`.
+
+* Specifying the namespace by way of the label
+  ``k8s:io.kubernetes.pod.namespace`` in the ``fromEndpoints`` and
+  ``toEndpoints`` fields is supported as described in :ref:`example_cnp_egress_to_kube_system`.
+  However, specifying the namespace in the ``endpointSelector`` is prohibited,
+  as it would violate the namespace isolation principle of Kubernetes. The
+  ``endpointSelector`` always applies to pods of the namespace 
+  associated with the `CiliumNetworkPolicy` resource itself.
+
+* Using namespace-specific information like
+  ``io.cilium.k8s.namespace.labels`` within a ``fromEndpoints`` or
+  ``toEndpoints`` is supported only for a `CiliumClusterwideNetworkPolicy`
+  and not a `CiliumNetworkPolicy`. The `CiliumNetworkPolicy` is a namespaced
+  resource and therefore doesn't have any conception of namespaces.
```

Comment:
Let's do away with the list altogether and let the section heading do some lifting. Also some typo fixes and clarity nits for active voice.
```suggestion
Depending on how a policy is defined and created, Kubernetes namespaces are automatically taken into account.

Network policies imported directly with the :ref:`api_ref` apply to all
namespaces unless a namespace selector is specified as described in
:ref:`example_cnp_ns_boundaries`.

Network policies created and imported as `CiliumNetworkPolicy` CRD and
`NetworkPolicy` apply within the namespace. In other words, the policy **only** applies
to pods within that namespace. It's possible, however, to grant access to and
from pods in other namespaces as described in :ref:`example_cnp_across_ns`.

Specifying the namespace by way of the label
``k8s:io.kubernetes.pod.namespace`` in the ``fromEndpoints`` and
``toEndpoints`` fields is supported as described in 
:ref:`example_cnp_egress_to_kube_system`.
However, Kubernetes prohibits specifying the namespace in the ``endpointSelector``,
as it would violate the namespace isolation principle of Kubernetes. The
``endpointSelector`` always applies to pods of the namespace 
associated with the `CiliumNetworkPolicy` resource itself.

Using namespace-specific information like
``io.cilium.k8s.namespace.labels`` within a ``fromEndpoints`` or
``toEndpoints`` is supported only for a `CiliumClusterwideNetworkPolicy`
and not a `CiliumNetworkPolicy`. The `CiliumNetworkPolicy` is a namespaced
resource and therefore doesn't have any concept of namespaces.
```